circuit/src/threshold/native.rs

```diff
@@ -27,11 +36,20 @@ impl<const NUM_LIMBS: usize, const POWER_OF_TEN: usize> Threshold<NUM_LIMBS, POW
 		let num = ratio.numer();
 		let den = ratio.denom();
 
-		let num_decomposed =
+		let mut num_decomposed =
 			decompose_big_decimal::<Fr, NUM_LIMBS, POWER_OF_TEN>(num.to_biguint().unwrap());
-		let den_decomposed =
+		let mut den_decomposed =
 			decompose_big_decimal::<Fr, NUM_LIMBS, POWER_OF_TEN>(den.to_biguint().unwrap());
 
+		// Apply the scaling to `num` and `den`
+		let num_last_non_zero_idx = find_last_non_zero_index(&num_decomposed);
+		let den_last_non_zero_idx = find_last_non_zero_index(&den_decomposed);
+		let limb_idx = num_last_non_zero_idx
+			.unwrap_or_default()
+			.max(den_last_non_zero_idx.unwrap_or_default());
```

Comment:
We should check if `num_last_non_zero_idx == den_last_non_zero_idx` otherwise we cant do threshold checks

Reply:
Another constraint could be that last limb of denominator can't be zero, in any case

Reply:
Okay, I see the point.
Mostly, the `num_last_non_zero_idx` and `den_last_non_zero_idx` are the same. 
But, there could be the case that they are different, you mean.